.eslintrc.js

```diff
@@ -271,53 +272,26 @@ module.exports = {
         'import/no-restricted-paths': [
             'error',
             {
-                basePath: './src',
                 zones: [
                     {
-                        target: './intellisense/**/*.ts',
-                        from: './test',
-                        message:
-                            'Importing test modules from ./src/test into extension code in .src/intellisense, is not allowed.'
+                        target: './src/**[!test]**/**/*.ts',
+                        from: './src/test/**/*.ts',
+                        message: 'Importing test modules from ./src/test into extension code is not allowed.'
                     },
                     {
-                        target: './interactive-window/**/*.ts',
-                        from: './test',
-                        message:
-                            'Importing test modules from ./src/test into extension code in .src/interactive-window, is not allowed.'
+                        target: './src/**[!test]**/**/*[!.node].ts',
```

Comment:
These are where the magic happens. Basically it says:

if a file that is not in the test folder and doesn't have '.node' in it (target)
is importing from a file with '.node.ts' in it (from)
throw an error saying the message.

Reply:
The rest of the changes are just dealing with the fallout from this rule and my custom no node builtins rule being enabled.